### PR TITLE
Add Map Object and Card Interface

### PR DESCRIPTION
**Pull request by kingavatar**
## Issue that this pull request solves

 Closes: 

## Proposed changes
Adding Map object which handles cards inner data and removing older plugins

## Types of changes

_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (Documentation content changed)
- [ ] Other (please describe): 

## Checklist

_Put an `x` in the boxes that apply_

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] My changes does not break the current system and it passes all the current test cases.

## Screenshots

Please attach the screenshots of the changes made in case of change in user interface
image

## Other information

Any other information that is important to this pull request

### CODE REVIEW

File: src/app/components/new-board/new-board.component.ts [hunk 1]
```diff
@@ -608,9 +476,10 @@ export class NewBoardComponent implements OnInit {
 
   // Save board data
   saveData() {
-    const boardTitle = document.getElementById('cb-title').innerHTML.trim();
-    const boardlData = document.getElementById('main-box').innerHTML.trim();
-    this.apiService.saveBoardData(boardTitle, boardlData);
+    // const boardTitle = document.getElementById('cb-title').innerHTML.trim();
```
Comment (Kajol-Kumari) [inline src/app/components/new-board/new-board.component.ts:479]: Why do we need to comment these lines here?

File: src/app/components/new-board/new-board.component.ts [hunk 2]
```diff
@@ -423,158 +443,6 @@ export class NewBoardComponent implements OnInit {
         this.AddPdfRenderComponent.addPdfRenderToolBox(uid, event.target.files[0], this.reader);
       });
 
-      // Add code snippet
```
Comment (Kajol-Kumari) [inline src/app/components/new-board/new-board.component.ts:426]: why are we removing this? I am seeing few unexpected behaviour :-
1. Image plugin isn't working
2. Code snippet plugin is providing the option for choosing the URL (If I am not wrong that should appear for image plugin)
3. Graph plugin isn't behaving as expected for json input file.

Are these the the bugs or only happening with this PR?

**Review by Kajol-Kumari — COMMENTED** [review]
@kingavatar Please address the comments and it would be great if you could please add the GIF of the feature you added in this PR.

File: src/app/components/new-board/new-board.component.ts (hunk 2 above)
Comment (kingavatar) [reply src/app/components/new-board/new-board.component.ts:426]: 1. Okay we were planning to remove the image plugin and code snippet plugin. Corresponding changes in html code should be made.
2. Code snippet plugin was redundant before so is being removed.
3. file upload plugin can be used for image plugin and pdf upload functionality.
4. Graph plugin feature didn't know it was breaking. i have to confirm with @goliakshay357 .

File: src/app/components/new-board/new-board.component.ts (hunk 1 above)
Comment (kingavatar) [reply src/app/components/new-board/new-board.component.ts:479]: The saveData function is resolved in the upcoming PR by @goliakshay357  using the Map object created in this PR to solve #171 and #192 . These lines will be rewritten as of now so to indicate their position it has been commented out as of now.

**Comment by kingavatar** [comment]
> @kingavatar Please address the comments and it would be great if you could please add the GIF of the feature you added in this PR.

As of now as this PR is removing existing code and adding extra variables to handle data a GIF is not provided. Will do so for now on.

**Pull request merged**